test/unit/views/payment-method-view.js

```diff
@@ -204,6 +205,80 @@ describe('PaymentMethodView', function () {
     });
   });
 
+  describe('selecting vaulted payment methods', function () {
+    beforeEach(function () {
+      this.client = fake.client();
+      this.model = fake.model();
+      this.sandbox.stub(analytics, 'sendEvent');
+    });
+
+    it('sends analytic event when vaulted card selected', function () {
+      var view = new PaymentMethodView({
+        client: this.client,
+        model: this.model,
+        strings: strings,
+        paymentMethod: {
+          type: 'CreditCard',
+          nonce: 'nonce',
+          vaulted: true,
+          details: {
+            lastFour: 1111
+          }
+        }
+      });
+
+      view._choosePaymentMethod();
+
+      expect(analytics.sendEvent.withArgs(this.client, 'vaulted-card.select')).to.be.calledOnce;
+    });
+
+    it('sends analytic event when vaulted paypal selected', function () {
+      var view = new PaymentMethodView({
+        client: this.client,
+        model: this.model,
+        strings: strings,
+        paymentMethod: {
+          type: 'PayPalAccount',
+          vaulted: true,
+          details: {
+            email: 'test@example.com'
+          }
+        }
+      });
+
+      view._choosePaymentMethod();
+
+      expect(analytics.sendEvent.withArgs(this.client, 'vaulted-paypal.select')).to.be.calledOnce;
+    });
+
+    it('does not send analytic event when no vault payment selected', function () {
+      expect(analytics.sendEvent.withArgs(this.client, 'vaulted-card.select')).to.not.be.called;
+    });
+
+    it('does not send analytic event when non-vaulted paypal selected', function () {
```

Comment:
can we make these `it` blocks a little more human readable. Ideally, if you concatenated the `describe` block with each `it` block, it should make sense. The string in the `describe` function is the noun that `it` is referring to.

So in this case, it reads: `selecting vaulted payment methods does not send analytic event when non-vaulted paypal selected`

A more readable version might be: `selecting vaulted payment methods does not send an analytic event when a PayPal payment method that is not vaulted is selected`

Reply:
[Fixes here](https://github.com/braintree/braintree-web-drop-in/pull/508/commits/d33aadd655c0e1fc1d6f4d347849dd16e364c992)